spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/neo4j/Neo4jDriverProperties.java

```diff
@@ -0,0 +1,497 @@
+	 * schemes. Defaults to bolt://localhost:7687.
+	 */
+	private URI uri = URI.create("bolt://localhost:7687");
+
+	/**
+	 * The authentication the driver is supposed to use. Maybe null.
+	 */
+	private Authentication authentication = new Authentication();
+
+	/**
+	 * The configuration of the connection pool.
+	 */
+	private PoolSettings pool = new PoolSettings();
+
+	/**
+	 * Detailed configuration of the driver.
+	 */
+	private DriverSettings config = new DriverSettings();
+
+	public URI getUri() {
+		return this.uri;
+	}
+
+	public void setUri(URI uri) {
+		this.uri = uri;
+	}
+
+	public Authentication getAuthentication() {
+		return this.authentication;
+	}
+
+	public void setAuthentication(Authentication authentication) {
+		this.authentication = authentication;
+	}
+
+	public PoolSettings getPool() {
+		return this.pool;
+	}
+
+	public void setPool(PoolSettings pool) {
+		this.pool = pool;
+	}
+
+	public DriverSettings getConfig() {
+		return this.config;
+	}
+
+	public void setConfig(DriverSettings config) {
+		this.config = config;
+	}
+
+	public AuthToken getAuthToken() {
+		return this.authentication.asAuthToken();
+	}
+
+	public Config asDriverConfig() {
+
+		Config.ConfigBuilder builder = Config.builder();
+		this.pool.applyTo(builder);
+		String scheme = (this.uri != null) ? this.uri.getScheme() : "bolt";
+		this.config.applyTo(builder, isSimpleScheme(scheme));
+
+		return builder.withLogging(new Neo4jSpringJclLogging()).build();
+	}
+
+	static boolean isSimpleScheme(String scheme) {
+
+		String lowerCaseScheme = scheme.toLowerCase(Locale.ENGLISH);
+		try {
+			Scheme.validateScheme(lowerCaseScheme);
+		}
+		catch (IllegalArgumentException ex) {
+			throw new IllegalArgumentException(String.format("'%s' is not a supported scheme.", scheme));
+		}
+
+		return lowerCaseScheme.equals("bolt") || lowerCaseScheme.equals("neo4j");
+	}
+
+	public static class Authentication {
+
+		/**
+		 * The login of the user connecting to the database.
+		 */
+		private String username;
+
+		/**
+		 * The password of the user connecting to the database.
+		 */
+		private String password;
+
+		/**
+		 * The realm to connect to.
+		 */
+		private String realm;
+
+		/**
+		 * A kerberos ticket for connecting to the database. Mutual exclusive with a given
+		 * username.
+		 */
+		private String kerberosTicket;
+
+		public String getUsername() {
+			return this.username;
+		}
+
+		public void setUsername(String username) {
+			this.username = username;
+		}
+
+		public String getPassword() {
+			return this.password;
+		}
+
+		public void setPassword(String password) {
+			this.password = password;
+		}
+
+		public String getRealm() {
+			return this.realm;
+		}
+
+		public void setRealm(String realm) {
+			this.realm = realm;
+		}
+
+		public String getKerberosTicket() {
+			return this.kerberosTicket;
+		}
+
+		public void setKerberosTicket(String kerberosTicket) {
+			this.kerberosTicket = kerberosTicket;
+		}
+
+		AuthToken asAuthToken() {
```

Comment:
Although a number of other properties have the same structure, we are trying to make sure `@ConfigurationProperties`-annotated type are as simple as possible and do not contain any logic. Please move this to the configuration if possible.